pkg/util/kubeobjects/pod/query.go

```diff
@@ -0,0 +1,38 @@
+package pod
+
+import (
+	"github.com/Dynatrace/dynatrace-operator/pkg/logd"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/hasher"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/internal/query"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func Query(kubeClient client.Client, kubeReader client.Reader, log logd.Logger) query.Generic[*corev1.Pod, *corev1.PodList] {
+	return query.Generic[*corev1.Pod, *corev1.PodList]{
+		Target:     &corev1.Pod{},
+		ListTarget: &corev1.PodList{},
+		ToList: func(sl *corev1.PodList) []*corev1.Pod {
+			out := []*corev1.Pod{}
+			for _, s := range sl.Items {
+				out = append(out, &s)
+			}
+
+			return out
+		},
+		IsEqual:     IsEqual,
+		IsImmutable: IsImmutable,
+
+		KubeClient: kubeClient,
+		KubeReader: kubeReader,
+		Log:        log,
+	}
+}
+
+func IsEqual(current, desired *corev1.Pod) bool {
```

Comment:
do we need <kind>.`Is*` functions  to be public?

Reply:
good point, not really, in some cases it might make sense, if it would be useable in all cases, but most of these are very specific checks, so making them private makes sense